### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by StevBL**
### Feat and Fix commits

1. **Feat**: config `.editorconfig`
2. **Feat**: config `ESLint (airbnb-base)`
3. **Fix**: configure the `indentation (2)`
4. **Fix**: use `const `in `DaysOfCode/classVsInstance.js`
5. **Fix**: use use better variable names in `DaysOfCode/dataTypes.js`
6. **Fix**: convert `DaysOfCode/loops.js` from `String `to `array`
7. **Fix**:  use `join('\n')` in `DaysOfCode/loops.js`

### CODE REVIEW

**Comment by StevBL** [comment]
### Feat and Fix commits

1. **Feat**: config `.editorconfig`
2. **Feat**: config `ESLint (airbnb-base)`
3. **Fix**: configure the `indentation (2)`
4. **Fix**: use const in `DaysOfCode/classVsInstance.js`
5. **Fix**: use use better variable names in `DaysOfCode/dataTypes.js`
6. **Fix**: convert `DaysOfCode/loops.js` from `String `to `array`
7. **Fix**: use` join('\n')` in `DaysOfCode/loops.js`
8. **Fix**: use `array.map` in `DaysOfCode/loops.js`

File: DaysOfCode/classVsInstance.js [hunk 1]
```diff
@@ -1,60 +1,71 @@
...
-    let age = 0;
-    // Add some more code to run some checks on initialAge
-    if (initialAge < 0) {
-        console.log('Age is not valid, setting age to 0.');
-        age = 0;
-    } else {
-        age = initialAge;
+  const age = 0;
+
+  function setAge(newAge) {
```
Comment (RobertBaron) [inline DaysOfCode/classVsInstance.js]: I am not a big fan of declaring functions inside functions please take this functions away., and have smaller functions, also please configure your eslint to not allow you to have you functions with more than 40 lines of code..

File: DaysOfCode/classVsInstance.js [hunk 2]
```diff
@@ -1,60 +1,71 @@
...
+    this.age = newAge;
+  }
+  function getAge() {
+    return this.age;
```
Comment (RobertBaron) [inline DaysOfCode/classVsInstance.js]: not a fan either, you have a raw function here, if you want to act like a class, use es6 classes..

File: DaysOfCode/classVsInstance.js [hunk 3]
```diff
@@ -1,60 +1,71 @@
...
+
+  // Add some more code to run some checks on initialAge
+  if (initialAge < 0) {
+    console.log('Age is not valid, setting age to 0.');
+    setAge(0);
+  } else {
+    setAge(initialAge);
+  }
+
+  this.amIOld = () => {
```
Comment (RobertBaron) [inline DaysOfCode/classVsInstance.js:35]: let's have this as a separate function.

File: DaysOfCode/classVsInstance.js [hunk 4]
```diff
@@ -1,60 +1,71 @@
+/* eslint-disable no-unused-vars */
```
Comment (RobertBaron) [inline DaysOfCode/classVsInstance.js:1]: please add the problem description as a comment in the top of the file

File: DaysOfCode/dataTypes.js [hunk 5]
```diff
@@ -8,41 +8,36 @@ let inputStdinArray = '';
...
 
-    // Print the sum of the double variables on a new line.
-    console.log((secD + d).toFixed(1));
-
-    // Concatenate and print the String variables on a new line
-    // The 's' variable above should be printed first.
-    console.log(s + secS);
+  // Concatenate and print the String variables on a new line
+  // The 's' variable above should be printed first.
+  console.log(thirdValue + thirdReadLine);
```
Comment (RobertBaron) [inline DaysOfCode/dataTypes.js]: use string literals to concatenate, using + can lead to errors.

File: DaysOfCode/dataTypes.js [hunk 6]
```diff
@@ -8,41 +8,36 @@ let inputStdinArray = '';
...
+  const firtsValue = 4;
+  const secValue = 4.0;
+  const thirdValue = 'HackerRank ';
 
-    // Declare second integer, double, and String variables.
-    let secI;
-    let secD;
-    let secS;
+  // Read and save an integer, double, and String to your variables.
+  const firtsReadLine = parseInt(readLine());
```
Comment (RobertBaron) [inline DaysOfCode/dataTypes.js:31]: is this code generated by hacker rank?

File: DaysOfCode/operators.js [hunk 7]
```diff
@@ -7,35 +7,35 @@ let inputString = '';
...
 // Complete the solve function below.
 function solve(meal_cost, tip_percent, tax_percent) {
-    let meal = parseFloat(meal_cost).toFixed(2);
-    let totTip = (meal * (parseFloat(tip_percent) / 100)).toFixed(2);
-    let totTax = (meal * (parseFloat(tax_percent) / 100)).toFixed(2);
-    let resul = Math.round(parseFloat(meal) + parseFloat(totTip) + parseFloat(totTax));
+  let meal = parseFloat(meal_cost).toFixed(2);
+  let totTip = (meal * (parseFloat(tip_percent) / 100)).toFixed(2);
+  let totTax = (meal * (parseFloat(tax_percent) / 100)).toFixed(2);
+  let resul = Math.round(parseFloat(meal) + parseFloat(totTip) + parseFloat(totTax));
```
Comment (RobertBaron) [inline DaysOfCode/operators.js]: why are you converting the totTip to string and then parsing it to a float later?

File: DaysOfCode/operators.js [hunk 8]
```diff
@@ -7,35 +7,35 @@ let inputString = '';
...
 }
 
 // Complete the solve function below.
 function solve(meal_cost, tip_percent, tax_percent) {
-    let meal = parseFloat(meal_cost).toFixed(2);
-    let totTip = (meal * (parseFloat(tip_percent) / 100)).toFixed(2);
-    let totTax = (meal * (parseFloat(tax_percent) / 100)).toFixed(2);
-    let resul = Math.round(parseFloat(meal) + parseFloat(totTip) + parseFloat(totTax));
+  let meal = parseFloat(meal_cost).toFixed(2);
+  let totTip = (meal * (parseFloat(tip_percent) / 100)).toFixed(2);
```
Comment (RobertBaron) [inline DaysOfCode/operators.js]: this is too simple of a case, but in some other cases a calc percentage function could be useful. specially when doing unit tests..

**Review by RobertBaron — CHANGES_REQUESTED** [review]

**Review by RobertBaron — APPROVED** [review]
this code works, you don't need to resolve those comments, let's improve other the next exercises, this PR is getting kinda of big.

File: DaysOfCode/dataTypes.js (hunk 6 above)
Comment (StevBL) [reply DaysOfCode/dataTypes.js:31]: yes, this code:
```
  var i = 4
  var d = 4.0
  var s = "HackerRank "

// Read and save an integer, double, and String to your variables.
  const firtsReadLine = parseInt(readLine());
  const secReadLine = parseFloat(readLine());
  const thirdReadLine = readLine();

```
but my repository use this code:

```
  const firtsValue = 4;
  const secValue = 4.0;
  const thirdValue = 'HackerRank ';

  // Read and save an integer, double, and String to your variables.
  const firtsReadLine = parseInt(readLine());
  const secReadLine = parseFloat(readLine());
  const thirdReadLine = readLine();
```

**Pull request merged**